src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateConsentedSubmittedHandlerTest.java

```diff
@@ -37,38 +41,73 @@ public class SolicitorCreateConsentedSubmittedHandlerTest {
     private FinremCaseDetailsMapper finremCaseDetailsMapper;
 
-    @Before
+    @BeforeEach
     public void setup() {
         handler =  new SolicitorCreateConsentedSubmittedHandler(finremCaseDetailsMapper, assignApplicantSolicitorService,
             createCaseService);
     }
 
     @Test
-    public void givenACcdCallbackSolicitorCreateConsentedCase_WhenCanHandleCalled_thenHandlerCanHandle() {
+    void givenACcdCallbackSolicitorCreateConsentedCase_WhenCanHandleCalled_thenHandlerCanHandle() {
         assertThat(handler
                 .canHandle(CallbackType.SUBMITTED, CaseType.CONSENTED, EventType.SOLICITOR_CREATE),
             is(true));
     }
 
     @Test
-    public void givenACcdCallbackAboutToSubmit_WhenCanHandleCalled_thenHandlerCanNotHandle() {
+    void givenACcdCallbackAboutToSubmit_WhenCanHandleCalled_thenHandlerCanNotHandle() {
         assertThat(handler
                 .canHandle(CallbackType.ABOUT_TO_SUBMIT, CaseType.CONTESTED, EventType.SOLICITOR_CREATE),
             is(false));
     }
 
     @Test
-    public void givenACcdCallbackSolicitorCreateContestedCase_WhenHandle_thenAddSupplementary() {
-        FinremCallbackRequest callbackRequest = buildFinremCallbackRequest();
-        handler.handle(callbackRequest, AUTH_TOKEN);
+    void givenACcdCallbackSolicitorCreateConsentedCase_WhenHandleWithAppRepresentation_thenAddSupplementaryAndAppRep() {
+        FinremCallbackRequest callbackRequest = buildFinremCallbackRequest(YesOrNo.YES);
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> response = handler.handle(callbackRequest, AUTH_TOKEN);
 
-        verify(createCaseService, times(1)).setSupplementaryData(eq(callbackRequest), any());
+        verify(createCaseService, times(1)).setSupplementaryData(callbackRequest, AUTH_TOKEN);
+        verify(assignApplicantSolicitorService, times(1))
+            .setApplicantSolicitor(callbackRequest.getCaseDetails(), AUTH_TOKEN);
+
+        assertTrue(response.getErrors().isEmpty());
     }
 
-    private FinremCallbackRequest buildFinremCallbackRequest() {
-        FinremCaseData caseData = FinremCaseData.builder().build();
+    @Test
+    void givenACcdCallbackSolicitorCreateConsentedCase_WhenHandleWithAppRepresentationError_thenAddSupplementaryAndHandleException() {
+        FinremCallbackRequest callbackRequest = buildFinremCallbackRequest(YesOrNo.YES);
+
+        doThrow(new AssignCaseAccessException("Exception message"))
+            .when(assignApplicantSolicitorService)
+            .setApplicantSolicitor(any(), any());
+
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> response = handler.handle(callbackRequest, AUTH_TOKEN);
+
+        verify(createCaseService, times(1)).setSupplementaryData(callbackRequest, AUTH_TOKEN);
+        verify(assignApplicantSolicitorService, times(1))
+            .setApplicantSolicitor(callbackRequest.getCaseDetails(), AUTH_TOKEN);
+
+        assertTrue(response.getErrors().contains("Exception message"));
```

Comment:
should we have `assertEquals(1, response.getErrors().size());` too?

Reply:
Yeah not a bad shout